api/v1alpha4/machinehealthcheck_types.go

```diff
@@ -56,6 +56,8 @@ type MachineHealthCheckSpec struct {
 
 	// Machines older than this duration without a node will be considered to have
 	// failed and will be remediated.
+	// If not set, this value is defaulted to 10 minutes.
+	// If you wish to disable this feature, set the value explicitly to 0.
 	// +optional
 	NodeStartupTimeout *metav1.Duration `json:"nodeStartupTimeout,omitempty"`
```

Comment:
Should we drop the pointer? It seems like this value is defaulted, and is omitted when `nil`, although `nil` isn't a valid value. If we set it to `0`, we disable it and we can also drop the `omitempty`?